### PR TITLE
add restarting and pausing support

### PR DESCRIPTION
**Pull request by HarshNarayanJha**
I gave implemented restarting the game without quitting and pausing logics. I have also added a flag on `Diddi`, called `is_invicible`, which disables any interaction with the Fire (for debugging purposes only!)

### CODE REVIEW

**Comment by HarshNarayanJha** [comment]
Oh! I accidently forgot to run black/flake8 formatter. It might have some issue!?

**Comment by DiddiLeija** [comment]
Since you're a "first-time contributor" in this repo, I have to approve GitHub workflow runs.
And... seems like flake8 isn't happy :/

File: main.py [hunk 1]
```diff
@@ -188,10 +190,12 @@ def __init__(self, x=0, y=0):
             "l": ((8, 8), (16, 8), (24, 8)),
         }
         self.alive = True
+        self.active = False
 
     def update(self):
         if not self.alive:
             return
+        if not self.active: return
```
Comment (DiddiLeija) [inline main.py]: _Maybe_ this could fix the failure?

```suggestion
        if not self.active:
            return
```

File: main.py [hunk 2]
```diff
@@ -327,9 +335,39 @@ def setup_game(self):
...
+        self.winner = False
+        self.playing = False
+        self.player = None
+        self.paused = False
+        pyxel.cls(0)
+        scroll_x = 0
+        self.setup_game()
+
+    def pause_game(self):
+        if self.paused: return
```
Comment (DiddiLeija) [inline main.py]: Same here.

```suggestion
        if self.paused:
            return
```

File: main.py [hunk 3]
```diff
@@ -327,9 +335,39 @@ def setup_game(self):
...
+        self.paused = True
+        self.player.active = False
+
+    def resume_game(self):
+        if not self.paused: return
```
Comment (DiddiLeija) [inline main.py]: Hmm... Seems like flake8 doesn't like things like this (`E701 multiple statements on one line (colon)`).

```suggestion
        if not self.paused:
            return
```

**Review by DiddiLeija — COMMENTED** [review]
I'd suggest you these changes to get rid of flake8 issues. Haven't run local formatters yet, but I deduced the fix according to the flake8 error messages. Hope it helps :smile:

**Comment by DiddiLeija** [comment]
Anyway, thank you so much for providing an improvement like this!

File: main.py [hunk 4]
```diff
@@ -338,10 +376,13 @@ def update_game(self):
...
                         if isinstance(enemy, Fire):
-                            self.player.alive = False
-                            self.play_death_sound()
+                            if not self.player.is_invicible:
+                                self.player.alive = False
+                                self.play_death_sound()
                         else:
                             self.winner = True
+                            # ... To disable scrolling calculations
+                            self.player.alive = False
```
Comment (DiddiLeija) [inline main.py:388]: Ah, another great fix included! I just noticed yesterday, but haven't fixed it yet! :tada:

**Comment by DiddiLeija** [comment]
Hmm... Seems like now black is unhappy :/

File: main.py (hunk 1 above)
Comment (HarshNarayanJha) [reply main.py]: I like this kind of one liner personally `if <condition>: statement`, if it is only one statement, like return or print, but I guess flake8 doesn't

File: main.py (hunk 4 above)
Comment (HarshNarayanJha) [reply main.py:388]: Yes, the view was still scrolling after winning 😆

File: main.py (hunk 1 above)
Comment (DiddiLeija) [reply main.py]: Indeed. I'm also a bit more +1 to the one-liner, but we gotta respect flake8' style :man_shrugging:

**Comment by DiddiLeija** [comment]
Gonna try to run the formatters locally, to find a possible fix :)

**Comment by HarshNarayanJha** [comment]
What's the issue this time 💦 ?

**Comment by DiddiLeija** [comment]
Black re-formatted something on my local run. Gonna make a suggestion as soon as possible :+1:

File: main.py [hunk 5]
```diff
@@ -363,9 +407,17 @@ def draw_game(self):
...
+                i.draw()
+            if self.paused:
+                display_text(adjust_x(50), 40, "PAUSED!")
+                display_text(adjust_x(30), 50, "Press R to Restart!")
+                display_text(adjust_x(30), 60, "Press P to Resume!")
+                display_text(adjust_x(30), 70, "Press Q to Quit!")
         else:
             display_text(adjust_x(30), 30, "Oh no! You lost!")
-            display_text(adjust_x(0), 40, "Press Q to quit, and try again!")
+            display_text(adjust_x(15), 40, f"Press R to restart, and\n{' '*6}try again!")
```
Comment (DiddiLeija) [inline main.py]: This is what Black suggested me:

```suggestion
            display_text(
                adjust_x(15), 40, f"Press R to restart, and\n{' '*6}try again!"
            )
```

Comment (HarshNarayanJha) [reply main.py]: CRazy suggestions 😆 ... that's the 80 characters line width I guess...

**Comment by HarshNarayanJha** [comment]
Oh! Finally it did, Congrats 🎉

**Review by DiddiLeija — APPROVED** [review]
Yipee! Thank you @HarshNarayanJha! :tada:

I've just tested this, and looks great. I'm going to merge this, update the guide with the new controls, and launch the fixed dists (we're still in time).

Thanks again!

**Pull request merged**